app/scripts/lib/metrics.js

```diff
@@ -153,6 +159,66 @@ define(function (require, exports, module) {
       this._clearInactivityFlushTimeout();
     },
 
+    notifications: {
+      /* eslint-disable sorting/sort-object-props */
+      'flow.initialize': '_initializeFlowModel',
+      'flow.event': '_logFlowEvent'
+      /* eslint-enable sorting/sort-object-props */
+    },
+
+    /**
+     * Initialize the flow model. If it's already been initalized, do nothing.
+     * Initialization may fail if the required flow properties can't be found,
+     * either in the DOM or the resume token.
+     */
+    _initializeFlowModel () {
+      if (this._flowModel) {
+        return;
+      }
+
+      const flowModel = new Flow({
+        sentryMetrics: this._sentryMetrics,
```

Comment:
I already opened #4639 for exactly this reason. Didn't want to do it in this PR though.